packages/BetterPhpDocParser/PhpDocManipulator/PhpDocTypeChanger.php

```diff
@@ -69,7 +70,11 @@ public function changeReturnType(PhpDocInfo $phpDocInfo, Type $newType): void
         }
 
         // override existing type
-        $newPHPStanPhpDocType = $this->staticTypeMapper->mapPHPStanTypeToPHPStanPhpDocTypeNode($newType);
+        if ($newType instanceof  NeverType) {
+            $newPHPStanPhpDocType = new IdentifierTypeNode('never');
+        } else {
+            $newPHPStanPhpDocType = $this->staticTypeMapper->mapPHPStanTypeToPHPStanPhpDocTypeNode($newType);
+        }
```

Comment:
see https://github.com/rectorphp/rector-src/pull/296#discussion_r658815909 and followup comments, why we need this thing here

Reply:
I see. The better solution is to update `Rector\PHPStanStaticTypeMapper\TypeMapper\NeverTypeMapper` that handles this `NeverType`

Reply:
> I see. The better solution is to update `Rector\PHPStanStaticTypeMapper\TypeMapper\NeverTypeMapper` that handles this `NeverType`

I totally agree and had this change before. have a look at https://github.com/rectorphp/rector-src/runs/2914993097 what tests all start to fail after I had changed the `NeverTypeMapper` like I need it for this PR.

Reply:
I see. In that case there should be a `$kind` parameter added.

- if kind === property then `mixed`
- if kind === return then `never`

Reply:
> In that case there should be a $kind parameter added.

soo adding a new parameter to the contract?
https://github.com/rectorphp/rector-src/blob/ac05fca06fad6a79abe431237671821d8d555153/packages/PHPStanStaticTypeMapper/Contract/TypeMapperInterface.php#L21

Reply:
Yes, that's the right place :+1:

Reply:
working on this idea in https://github.com/rectorphp/rector-src/pull/306 to isolate it